File: examples/js/okex-transfer.js
"use strict";

const ccxt = require ('../../ccxt.js')

console.log ('CCXT version:', ccxt.version)
console.log ('This example requires CCXT version 1.54.92 or higher')

async function main () {

    const exchange = new ccxt.okex ({
        'apiKey': 'YOUR_API_KEY',
        'secret': 'YOUR_SECRET',
        'password': 'YOUR_API_KEY_PASSWORD',
    })

    await exchange.loadMarkets ()

    const code = 'USDT'
    let tradingBalance = await exchange.fetchFreeBalance (/* { 'type': 'trading' } */)
    let fundingBalance = await exchange.fetchFreeBalance ({ 'type': 'funding' })
    console.log ('Trading:', tradingBalance[code], code)
    console.log ('Funding:', fundingBalance[code], code)

    const oldVerboseMode = exchange.verbose
    exchange.verbose = process.argv.includes ('--verbose') || process.argv.includes ('-v')

    // https://www.okex.com/docs-v5/en/#rest-api-funding-funds-transfer
    //
    //     'spot' == '1'
    //     'futures' == '3',
    //     'margin' == '5',
    //     'swap' == '9',
    //     'option' == '12',
    //     'trading' == '18', // unified trading account
    //     'unified' == '18',
    //
    const from = 'trading'
    const to = 'funding'
    const amount = 1
    const transfer = await exchange.transfer (code, amount, from, to)
    console.log (transfer)

    exchange.verbose = oldVerboseMode

    tradingBalance = await exchange.fetchFreeBalance (/* { 'type': 'spot' } */)
    fundingBalance = await exchange.fetchFreeBalance ({ 'type': 'funding' })
    console.log ('Trading:', tradingBalance[code], code)
    console.log ('Funding:', fundingBalance[code], code)
}

main ()

File: examples/py/okex-transfer.py
import ccxt
from pprint import pprint


# make sure your version is the latest
print('CCXT Version:', ccxt.__version__)

exchange = ccxt.okex({
    'apiKey': 'YOUR_API_KEY',
    'secret': 'YOUR_API_SECRET',
    'password': 'YOUR_API_PASSWORD',
})

exchange.load_markets()

# https://github.com/ccxt/ccxt/wiki/Manual#implicit-api-methods
# https://github.com/ccxt/ccxt/wiki/Manual#passing-parameters-to-api-methods
# uncomment to see all available methods
# pprint(dir(exchange))

code = 'BTC'
currency = exchange.currency(code)


try:
    response = exchange.account_post_transfer({
        'currency': currency['id'],
        'amount': '0.1',
        # 'type': '0',  # 0 transfer between accounts, 1 main to sub_account, 2 sub_account to main
        'from': '6',  # 1 spot, 3 futures, 5 margin, 6 funding account, 9 swap, 12 option
        'to': '1',  # 1 spot, 3 futures, 5 margin, 6 funding account, 9 swap, 12 option
        # 'sub_account': 'name_of_sub_account',  # when type is 1 or 2 sub_account is required
        # 'instrument_id': 'String',  # margin trading pair of token or underlying of USDT-margined futures transferred out, such as: btc-usdt. Limited to trading pairs available for margin trading or underlying of enabled futures trading.
        # 'to_instrument_id': 'String'  # margin trading pair of token or underlying of USDT-margined futures transferred in, such as: btc-usdt. Limited to trading pairs available for margin trading or underlying of enabled futures trading.
    })
    pprint(response)
except Exception as e:
    print(type(e).__name__, str(e))
